File: admin/src/test/java/com/beat/admin/user/facade/AdminUserFacadeTest.java
package com.beat.admin.user.facade;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

import java.util.List;

import org.junit.jupiter.api.Test;

import com.beat.admin.user.application.query.AdminUserQueryService;
import com.beat.admin.user.application.result.AdminUserResults;

class AdminUserFacadeTest {

	@Test
	void facadeDelegatesUserScenarioToQueryService() {
		AdminUserQueryService queryService = mock(AdminUserQueryService.class);
		AdminUserFacade adminUserFacade = new AdminUserFacade(queryService);
		when(queryService.findAllUsers(1L)).thenReturn(new AdminUserResults(List.of()));

		adminUserFacade.checkMemberAndFindAllUsers(1L);

		verify(queryService).findAllUsers(1L);
	}
}

File: admin/src/test/kotlin/com/beat/admin/user/facade/AdminUserFacadeTest.kt
package com.beat.admin.user.facade

import com.beat.admin.user.application.query.AdminUserQueryService
import com.beat.admin.user.application.result.AdminUserResults
import org.junit.jupiter.api.Test
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

class AdminUserFacadeTest {

    @Test
    fun facadeDelegatesUserScenarioToQueryService() {
        val queryService = mock(AdminUserQueryService::class.java)
        val adminUserFacade = AdminUserFacade(queryService)
        `when`(queryService.findAllUsers(1L)).thenReturn(AdminUserResults(emptyList()))

        adminUserFacade.checkMemberAndFindAllUsers(1L)

        verify(queryService).findAllUsers(1L)
    }
}